### PR TITLE
feat: limit access to priority queues

### PR DESCRIPTION
**Pull request by sileht**

### CODE REVIEW

File: mergify_engine/actions/merge/helpers.py [hunk 1]
```diff
@@ -87,16 +89,31 @@ def get_queue_summary(q):
...
         try:
             fancy_priority = PriorityAliases(priority).name
         except ValueError:
             fancy_priority = priority
         formatted_pulls = ", ".join((f"#{p}" for p in grouped_pulls))
         summary += f"\n* {formatted_pulls} (priority: {fancy_priority})"
+
+    subscription = sub_utils.get_subscription(
+        utils.get_redis_for_cache(), q.installation_id,
+    )
```
Comment (jd) [inline mergify_engine/actions/merge/helpers.py:112]: The subscription is already attached to the context.

File: mergify_engine/actions/merge/queue.py [hunk 2]
```diff
@@ -103,13 +107,24 @@ def _add_pull(self, pull_number, priority, update=False):
...
 
-    def add_pull(self, pull_number, config):
-        self._add_pull(pull_number, config["priority"])
+    def add_pull(self, ctxt, config):
+        config = config.copy()
+        config["effective_priority"] = config["priority"]
+
+        subscription = sub_utils.get_subscription(
+            utils.get_redis_for_cache(), self.installation_id,
+        )
```
Comment (jd) [inline mergify_engine/actions/merge/queue.py:116]: ditto

**Comment by jd** [comment]
This also needs to update the documentation.

File: doc/source/global-substitutions.rst [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.. |pro plan tag| raw:: html
+
+   <div style="text-align:right"><b><i><a href="https://mergify.io/pricing" target="_blank">Pro Plan</a> subscribers only 🦾</i></b></div>
```
Comment (jd) [inline doc/source/global-substitutions.rst:3]:
```suggestion
   https://www.notion.so/juldanjou/Add-a-squash-command-6f6dd84ef62e40878b1414e674594323
```

Screenshot 2020-06-03 at 18 54 32

Comment (sileht) [reply doc/source/global-substitutions.rst:3]: Great!

**Pull request merged**